### PR TITLE
Fix being able to select non builders hut using resource scroll

### LINKED ISSUE #407: Linking resource scroll to non builders hut
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my issue is not covered there.
- [x] I made sure that **this change request is not a duplicate of any existing change request.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: minecolonies-0.14.134-ALPHA-universal.jar
- Related Mods and their Versions:

## Is your feature request related to a problem?
-

## Enhancement or Change description
Currently it's possible to link a resource scroll to a non builders hut, instead of saying something like "Please select a builders hut to link this resource scroll to" it actually just says "Linked resources for Builder's Hut". If you then proceed to use the resource scroll it says you have to bind it to a builders hut, small QoL change that could make it a bit better.

### Notes or related things
-

---
## Viewers

* Add a thumbs-up to the change request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Thodor12**
Closes ldtteam/minecolonies-features#407
Closes #
Closes #

# Changes proposed in this pull request:
- Prevents players from linking a resource scroll to non builders hut buildings, change would include one new translation message to be added

Review please

(First PR, should I make multiple PR's to fix this identical issue on other versions or how does the process go?)

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/items/ItemResourceScroll.java [hunk 1]
```diff
@@ -70,14 +71,24 @@ public ActionResultType useOn(ItemUseContext ctx)
...
-            BlockPosUtil.write(compound, TAG_BUILDER, ((AbstractTileEntityColonyBuilding) entity).getPosition());
-
-            if (!ctx.getLevel().isClientSide)
-            {
-                LanguageHandler.sendPlayerMessage(ctx.getPlayer(),
-                  TranslationConstants.COM_MINECOLONIES_SCROLL_BUILDER_SET,
-                  ((AbstractTileEntityColonyBuilding) entity).getColony().getName());
+            final AbstractTileEntityColonyBuilding buildingEntity = (AbstractTileEntityColonyBuilding) entity;
+
+            if (buildingEntity.getBuilding() instanceof BuildingBuilder) {
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/items/ItemResourceScroll.java]: Please adjust the formatting to fit our style. You can find a formatting file in the documentation folder in the REPO.
Manual formatting would be then

{


}

**Review by Raycoms — CHANGES_REQUESTED** [review]
Just a small formatting problem

**Comment by Raycoms** [comment]
I'll port it over to the other versions myself through some cherrypicks

**Comment by Thodor12** [comment]
Regarding the TODO about the building names, is this proper name available somewhere?
I checked throughout the building entity and building class but the best I could find was the schematic name.

**Review by marchermans — DISMISSED** [review]

**Comment by Raycoms** [comment]
> Regarding the TODO about the building names, is this proper name available somewhere? I checked throughout the building entity and building class but the best I could find was the schematic name.

The building entry got a translation key.

**Review by marchermans — APPROVED** [review]

**Pull request merged**